src/pydiscourse/client.py

```diff
@@ -604,6 +605,87 @@ def delete_topic(self, topic_id, **kwargs):
 
         """
         return self._delete(f"/t/{topic_id}", **kwargs)
+    
+    def _topic_attachments_urls(self, slug, topic_id, **kwargs):
+        """
+
+        Args:
+            slug:
+            topic_id:
+            **kwargs:
+
+        Returns:
+            list of urls
+        """
+        topic = self.topic(slug, topic_id, **kwargs)
+        posts = [post['cooked'] for post in \
+                 topic['post_stream']['posts']]
+        attachments_urls = [ATTACHMENTS_REGEX.findall(post) for post in posts]
+        attachments_urls = {
+            group
+            for post in attachments_urls for match in post for group in match
+            if group
+        }
+
+        return attachments_urls
+
+    def topic_attachments_number(self, slug, topic_id, **kwargs):
+        """
+
+        Args:
+            slug:
+            topic_id:
+            **kwargs:
+
+        Returns:
+            number of attachments per topic
+        """
+        return len(self._topic_attachments_urls(slug, topic_id))
+
+    def topic_attachments(self, slug, topic_id, allowed_extensions=None, **kwargs):
+        """
+
```

Comment:
Please add a comment explaining what this function does (for example see https://github.com/pydiscourse/pydiscourse/blob/53be265b2345b2a294c79ec4c35d40992197f3a8/src/pydiscourse/client.py#L1370
In particular this method name gives no indication of if its for listing or uploading